realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -856,13 +872,42 @@ void notifyChangeListeners() {
             //FIXME: still waiting for Core to provide a fix
             //       for crash when calling _sync_if_needed on a cleared View.
             //       https://github.com/realm/realm-core/pull/1390
-            long version = table.sync();
+            long version;
+            try {
+                version = table.sync();
+            } catch (DeleteRealmListException e) {
+                // Although this RealmResults won't be updated anymore, it is good to give user a chance to do update.
+                version = TABLE_VIEW_VERSION_REALM_LIST_DELETED;
+                RealmLog.d("The parent RealmList has been deleted already.");
+            }
             if (currentTableViewVersion != version) {
                 currentTableViewVersion = version;
                 for (RealmChangeListener listener : listeners) {
                     listener.onChange();
                 }
             }
+            // No point to keep those listeners, just clear them.
+            if (version == TABLE_VIEW_VERSION_REALM_LIST_DELETED) {
```

Comment:
Is this a good idea? As the user are not aware they get removed? I would probably log it is happening.